clash-prelude/src/Clash/Promoted/Nat.hs

```diff
@@ -115,15 +117,39 @@ instance ShowX (SNat n) where
 withSNat :: KnownNat n => (SNat n -> a) -> a
 withSNat f = f SNat
 
+-- | Same as 'snatToInteger', but doesn't take term arguments. Example usage:
+--
+-- >>> natToInteger @5
+-- 5
+natToInteger :: forall n . KnownNat n => Integer
+natToInteger = snatToInteger (SNat @n)
+{-# INLINE natToInteger #-}
+
 -- | Reify the type-level 'Nat' @n@ to it's term-level 'Integer' representation.
 snatToInteger :: SNat n -> Integer
 snatToInteger p@SNat = natVal p
 {-# INLINE snatToInteger #-}
 
+-- | Same as 'snatToNatural', but doesn't take term arguments. Example usage:
+--
+-- >>> natToNatural @5
+-- 5
+natToNatural :: forall n . KnownNat n => Natural
```

Comment:
This is `GHC.TypeNats.natVal` (which is distinct from `GHC.TypeLits.natVal`)

Reply:
I completely missed the difference between `GHC.TypeNats.natVal` and `GHC.TypeLits.natVal`. It's not the same right? This function is ambiguous. I.e.,

```haskell
natVal :: forall n proxy . KnownNat n => proxy n -> Natural
```

vs

```haskell
natToNatural :: forall n. KnownNat n => Natural
```

Reply:
Is there any reason to prefer `natToNatural @5` over `natVal (Proxy @5)`?

Reply:
No `Proxy` clutter, clear difference between `GHC.TypeNats.natVal` and `GHC.TypeLits.natVal`. But it doesn't make a big difference.

Reply:
in the places where you've used these new functions, they can be rewritten using `GHC.TypeNats.natVal` like so:

```
xyz :: KnownNat n => Unsigned n
xyz = let r = case GHC.TypeNats.natVal r of { ... }  in r
```

since our numeric types are all proxies for `n`

Reply:
Personally I would say using `Proxy` is less clutter than defining a (technically) redundant function, although granted this doesn't matter too much because it doesn't leak into user-facing APIs.

:man_shrugging:

Reply:
Actually, that's why I wrote the functions in the first place. 

```haskell
xyz :: forall n. KnownNat n => Unsigned n
xyz = case natToNatural @n of { ... }
```

seems more obvious to me.

I'd be happy to revert it though if there are strong opinions on this.